lib/DxcSupport/WinAdapter.cpp

```diff
@@ -51,7 +51,15 @@ BSTR SysAllocStringLen(const OLECHAR *strIn, UINT ui) {
 
   return strOut;
 }
+//===--------------------------- BSTR Length ------------------------------===//
+unsigned int SysStringLen(const BSTR bstrString) {
+  if (!bstrString)
+    return 0;
 
+  uint32_t *blobIn = (uint32_t *)((uintptr_t)bstrString - sizeof(uint32_t));
+
+  return blobIn[0] / sizeof(OLECHAR);
```

Comment:
While I don't imagine BSTR is going to be anything but a pointer to OLECHARs any time soon, a slightly less implementation-dependent approach would be to use `sizeof(bstrString[0])`

Reply:
We added the length to (uintptr_t)bstrString - sizeof(uint32_t) in https://github.com/microsoft/DirectXShaderCompiler/pull/5942/files#diff-387db224f410dd80e8ea3a8ebec09badd12bcecc88132f28e816f9210e209a86R42